client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,289 @@
+
+    // Finally, build the table
+    let tbl = document.createElement('table');
+    tbl.style.height = this.get('height') + 'px';
+    tbl.style.width = this.get('width')+ 'px';
+    let numRows;
+    let numCols;
+    let currentRound = parseInt(this.get('currentRound'));
+
+    if (isNaN(currentRound) || currentRound <1 || currentRound > numRounds)
+      currentRound = null;
+    
+    if(this.get('playersInColumns')) {
+      // Compute total number of rows and columns in table
+      numCols = pScores.length + 1;
+      numRows = numRounds + 1 + (showTotals ? 1 : 0);
+
+      // Add header row
+      const names = pScores.map(x => x[0]);
+      names.unshift(this.get('roundLabel'));
+      tbl.innerHTML += '<tbody></tbody>';
+      const tr = this.addRowToTable($('tbody', tbl), names, 'td');
+      if(usePlayerColors)
+        for (let c=0; c<pScores.length; c++ ) {
+          tr.cells[c+1].style.backgroundColor = colors[c];
+          tr.cells[c+1].style.color = 'white';
+        }
+      // Add remaining rows
+      for( let r=1; r < numRows; r++ ) {
+        const pRow = pScores.map(x => x[r]);
+        pRow.unshift(rounds[r]);
+        const tr = this.addRowToTable($('tbody',tbl), pRow, 'td');
+        tr.cells[0].classList.add('firstCol');
+        if(r == currentRound)
+          for(let c=1; c < numCols; c++)
+            tr.cells[c].classList.add('currentRound');
+      }
+      if(showTotals)
+          for(let c=0; c < numCols; c++)
+            tbl.rows[numRows-1].cells[c].classList.add('totalsLine');
+    } else { // Scores are in rows
+      // Compute total number of rows and columns in table
+      numCols = numRounds + 1 + (showTotals ? 1 : 0);
+      numRows = pScores.length + 1;
+
+      // First row contains round names
+      const tr = this.addRowToTable(tbl, rounds, 'td');
+      for (let c=0; c < numCols; c++)
+        tr.cells[c].classList.add('firstRow');
+      // Remaining rows are one row per player.
+      for( let r=0; r < pScores.length; r++) {
+        const tr = this.addRowToTable(tbl, pScores[r], 'td');
+        tr.cells[0].classList.add('firstCol');
+        if(usePlayerColors) {
+          tr.cells[0].style.backgroundColor = colors[r];
+          tr.cells[0].style.color = 'white';
+        }
+      }
+      for(let r=1; r < numRows; r++) {
+        if(currentRound)
+          tbl.rows[r].cells[currentRound].classList.add('currentRound');
+        if(showTotals)
+          tbl.rows[r].cells[numCols-1].classList.add('totalsLine');
+      }
+    }
+
+    // Make top row, left column non-scrollable
+    for (let r=0; r<numRows; r++) 
+      tbl.rows[r].cells[0].classList.add('firstCol');
+    for (let c=0; c< numCols; c++)
+      tbl.rows[0].cells[c].classList.add('firstRow');
+
+    const myDom = this.domElement;
+    const tableElements = myDom.querySelectorAll('tr, td, th, table'); //This line and next suggested by ChatGPT
+    tableElements.forEach(element => element.remove());
```

Comment:
I thought about something like this. This will break if there is ever a second table as a child to the scoreboard. Unlikely, I agree, but I thought a better solution would be to give the table an id and just remove the dom element with that id. I'll leave it to Dave or Jan to comment.

Reply:
Also, I didn't think to try this until now, but you can remove 'tr, td, th' and just leave table.  That also works, but I won't tweak that.

Reply:
And you are correct of course.  After testing, if you make a scoreboard the parent of another scoreboard, then the child is scoreboard is removed (and the same would undoubtedly happen if any child element had a table).